### PR TITLE
Popupviewer -  maui

### PR DESCRIPTION
**Pull request by dotMorten**
PopupViewer for .NET MAUI

https://github.com/Esri/arcgis-maps-sdk-dotnet-toolkit/assets/1378165/ff4df025-dda1-43d8-8827-6e0c4e147770

Still outstanding issue: Finish full HTML rendering with clickable links.

### CODE REVIEW

File: src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupMediaView.cs [hunk 1]
```diff
@@ -102,30 +96,52 @@ private void UpdateImage()
...
+                    img.GestureRecognizers.Clear();
+                    var tapGesture = new TapGestureRecognizer();
+                    tapGesture.Tapped += (s, e) => _ = Launcher.LaunchUriAsync(linkUrl);
+                    img.GestureRecognizers.Add(tapGesture);
+#else
+                    img.Tag = linkUrl;
+                    if(img.Cursor != Cursors.Hand)
+                    {
+                        img.Cursor = Cursors.Hand;
+                        img.MouseLeftButtonDown += (s, e) => _ = Launcher.LaunchUriAsync((s as Image).Tag as Uri);
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupMediaView.cs:135]: This line produces build warnings:
> CS8604	Possible null reference argument for parameter `uri` in `Task<bool> Launcher.LaunchUriAsync(Uri uri)`.
> CS8602	Dereference of a possibly null reference.

File: src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+            }
+            if(parent is AttachmentsPopupElementView a && cell?.BindingContext is AttachmentViewModel vm)
+            {
+                a.OnAttachmentClicked(vm.Attachment);
+            }
+        }
+        
+
+#if WINDOWS
+        private bool _isSaveDialogOpen = false;
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs:137]: Dead code?

> CS0414	The field `AttachmentsPopupElementView._isSaveDialogOpen` is assigned but its value is never used

File: src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                }
+                catch
+                {
+                }
+            }
+        }
+
+        private class AttachmentViewModel : System.ComponentModel.INotifyPropertyChanged
+        {
+            private string? _filename;
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs:186]: Dead code?

> Warning	CS0169	The field 'AttachmentsPopupElementView.AttachmentViewModel._filename' is never used

File: src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+            public bool IsDownloading => !Attachment.IsLocal && Attachment.LoadStatus == LoadStatus.Loading;
+
+            private void CreateThumbnail()
+            {
+                if (Attachment.IsLocal || Attachment.LoadStatus == LoadStatus.Loaded)
+                {
+                    _thumbnail = ImageSource.FromStream(async (token) =>
+                    {
+                        var img = await Attachment.CreateThumbnailAsync(40, 40);
+                        return await img!.GetEncodedBufferAsync();
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/PopupViewer/AttachmentsPopupElementView.Maui.cs:253]: We should probably forward `token` to GetEncodedBufferAsync?

File: src/Samples/Toolkit.SampleApp.Maui/Samples/PopupViewerSample.xaml.cs [hunk 5]
```diff
@@ -3,56 +3,39 @@
...
+    {
+        public PopupViewerSample()
+        {
+            InitializeComponent();
+            mapView.Map = new Esri.ArcGISRuntime.Mapping.Map(new Uri("https://www.arcgis.com/home/item.html?id=9f3a674e998f461580006e626611f9ad"));
             mapView.GeoViewTapped += mapView_GeoViewTapped;
-        }  // Used in Callout to see feature details in PopupViewer
+        }
+
         private RuntimeImage InfoIcon => new RuntimeImage(new Uri("https://cdn3.iconfinder.com/data/icons/web-and-internet-icons/512/Information-256.png"));
```
Comment (mstefarov) [inline src/Samples/Toolkit.SampleApp.Maui/Samples/PopupViewerSample.xaml.cs:21]: The InfoIcon is no longer used -- can this be removed too?

File: src/Samples/Toolkit.SampleApp.Maui/Samples/PopupViewerSample.xaml.cs [hunk 6]
```diff
@@ -121,5 +104,26 @@ private Popup GetPopup(IEnumerable<IdentifyLayerResult> results)
...
+        {
+            popupPanel.IsVisible = false;
+        }
+
+        private void CloseButton_Click(object sender, EventArgs e)
+        {
+            popupPanel.IsVisible = false;
+        }
+
+        private void popupViewer_PopupAttachmentClicked(object sender, Esri.ArcGISRuntime.Toolkit.Maui.PopupAttachmentClickedEventArgs e)
```
Comment (mstefarov) [inline src/Samples/Toolkit.SampleApp.Maui/Samples/PopupViewerSample.xaml.cs:118]: Are these handlers supposed to be hooked up? They look unused now.

Comment (dotMorten) [reply src/Samples/Toolkit.SampleApp.Maui/Samples/PopupViewerSample.xaml.cs:118]: OnBackgroundTapped isn't used (will remove). The other two are hooked up in the xaml

**Review by mstefarov — APPROVED** [review]
Looks great. Well done!

**Pull request merged**